app/knights.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+
+class Knights:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list[dict],
+            weapon: dict,
+            potion: dict | None,
+            protection: int = 0
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.protection = protection
+
+    def arm_himself(self) -> None:
+        self.power += self.weapon["power"]
+
+    def putting_on(self) -> None:
+        for element in self.armour:
+            self.protection += element["protection"]
+
+    def application(self) -> None:
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:

```

Comment:
The code assumes that `self.potion['effect']` is a dictionary. It's safer to check if 'effect' exists in `self.potion` before accessing it to avoid potential KeyError exceptions.